File: packages/csharp/tests/Zyins/Fixtures.cs
// Shared test fixtures: canonical applicants matching the project's
// persona discipline (api-standards.md §13). John Doe is the
// primary; Jane Smith is the secondary.
namespace Sah.Sdk.Zyins.Tests;

internal static class Fixtures
{
    public static Applicant JohnDoe() => new()
    {
        Dob = "1962-04-18",
        Sex = Sex.Male,
        HeightInches = 70,
        WeightPounds = 195,
        State = "NC",
        Zip = "27514",
        NicotineUse = NicotineUsage.None,
    };

    public static Applicant JaneSmith() => new()
    {
        Dob = "1985-11-02",
        Sex = Sex.Female,
        HeightInches = 65,
        WeightPounds = 140,
        State = "CA",
        Zip = "94110",
        NicotineUse = NicotineUsage.None,
    };

    public static PrequalifyInput JohnDoePrequalifyInput() => new()
    {
        Applicant = JohnDoe(),
        Coverage = Coverage.ByFaceValue(50_000),
        Products = new[] { new Product("colonial-penn", "cp-senior-life") },
    };

    public const string SampleToken = "isa_test_4fjK2nQ7mX1aB8sR9pZ3";
    public const string SampleRequestId = "req_01HZK2N5GQR9T8X4B6FJW3Y1AS";
    public const string SampleSessionSecret = "test_session_secret_for_hmac_only";
}

File: packages/ts/tests/zyins/fixtures.ts
/**
 * Shared test fixtures for the Tier 3 ZyINS suite.
 *
 * Persona discipline (api-standards.md §"Persona discipline"): every test
 * uses the same primary applicant, the same auth, and the same product
 * selection. The recognition compounds when failures surface in CI logs.
 */

import type { AuthContext } from '../../src/zyins/auth';
import {
  type Applicant,
  Sex,
  NicotineDuration,
  Height,
  Weight,
} from '../../src/zyins/applicant';
import { Coverage } from '../../src/zyins/coverage';
import { Products, ProductSelection } from '../../src/zyins/product';

export const TEST_AUTH: AuthContext = {
  licenseKey: 'LIC-ABC-123',
  orderId: 'ORD-42',
  email: 'john.doe@acme-agency.com',
  deviceId: 'device-xyz-123',
};

export const TEST_APPLICANT: Applicant = {
  dob: '1962-04-18',
  sex: Sex.Male,
  height: Height.fromFeetInches(5, 10),
  weight: Weight.fromPounds(195),
  state: 'NC',
  nicotineUse: { lastUsed: NicotineDuration.Never },
};

export const TEST_COVERAGE = Coverage.faceValue(100_000);

const AETNA_ACCENDO = Products.Fex.AetnaAccendo;
if (!AETNA_ACCENDO) {
  throw new Error('Missing test fixture product: Products.Fex.AetnaAccendo');
}
export const TEST_PRODUCTS = ProductSelection.of([AETNA_ACCENDO]);

export const FIXED_CLOCK = (): number => 1_700_000_000_000;
